### ISSUE TITLE
Pyright fails to narrow types inside while loop

### ISSUE THREAD

**Issue by ethframe** [microsoft/pyright#2191]
**Describe the bug**
Under certain circuimstances, pyright fails to narrow types of variables in the body of a while loop.

**To Reproduce**

```python
from typing import Dict


class A:
    pass


class B(A):
    pass


def foo(v: A, s: Dict[B, A]) -> object:
    if not isinstance(v, B):
        return v
    u = s.get(v)
    while isinstance(u, B):
        v = u
        u = s.get(v)
    return v if u is None else u
```

produces the following error:

```
foo.py:18:13 - error: No overloads for "get" match the provided arguments
  Argument types: (B | A) (reportGeneralTypeIssues)
```

**Expected behavior**
No errors.

**VS Code extension or command-line**
pyright 1.1.162.

**Additional context**

If I replace `v` by `u` in the second call to `s.get`, the error disappears. This seems odd, because it does not change the behavior of the code - it just uses a different variable to access the same value. Mypy accepts both variants of the code.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.167, which I just published. It will also be included in the next release of pylance.
